src/App.tsx

```diff
@@ -1,110 +1,135 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const options = [3, 5, 10, 20];
+
+interface ItemsListProps {
+  itemL: string[];
+}
+
+interface PaginationProps {
+  pageCount: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const ItemList: React.FC<ItemsListProps> = ({ itemL }) => (
+  <ul>
+    {itemL.map(item => (
+      <li key={item} data-cy="item">
+        {item}
+      </li>
+    ))}
+  </ul>
+);
+
+export const Pagination: React.FC<PaginationProps> = ({
+  pageCount,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, pageCount).map(n => n);
+
+  return (
+    <ul className="pagination">
+      <li className={`page-item ${currentPage === 1 ? 'disabled' : ''}`}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => currentPage > 1 && onPageChange(currentPage - 1)}
+        >
+          «
+        </a>
+      </li>
+
+      {pages.map(page => (
+        <li
+          key={page}
+          className={`page-item ${currentPage === page ? 'active' : ''}`}
+          onClick={() => onPageChange(page)}
```

Comment:
Ensure that `onPageChange` is only called if the page actually changes. Consider adding a condition to check if `currentPage` is different from `page` before calling `onPageChange`.